test/e2e/network/service.go

```diff
@@ -3283,42 +3282,76 @@ var _ = SIGDescribe("ESIPP [Slow]", func() {
 			framework.Failf("Service HealthCheck NodePort still present")
 		}
 
-		endpointNodeMap, err := jig.GetEndpointNodes()
+		config := e2enetwork.NewNetworkingTestConfig(f, false, false)
+
+		epNodeNames := sets.NewString()
+		epNodes, err := jig.ListNodesWithEndpoint()
 		framework.ExpectNoError(err)
-		noEndpointNodeMap := map[string][]string{}
+		// map from name of nodes with endpoint to internal ip
+		endpointNodeMap := make(map[string]string)
+		// map from name of nodes without endpoint to internal ip
+		noEndpointNodeMap := make(map[string]string)
+		for _, node := range epNodes {
+			ips := e2enode.GetAddresses(node, v1.NodeInternalIP)
+			if len(ips) < 1 {
+				framework.Failf("No internal ip found for node %s", node.Name)
+			}
+			endpointNodeMap[node.Name] = ips[0]
+			epNodeNames.Insert(node.Name)
```

Comment:
Let's remove the `epNodeName` var completely - it has an entry for the same stuff as endpointNodeMap.

It seems to be used only in L3308, when you can simply have:
```
if _, ok := endpointNodeMap[n.Name]; !ok {
```

Reply:
nice catch!